liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -213,18 +279,30 @@ private static boolean isValidURL(String url) {
         }
     }
    
-    public static boolean isFoundWebAppElement(NodeList nodeList) {
-         
+    private static boolean isFoundAppConfigFromAppList(NodeList nodeList) {
+        return isFoundAppConfigFromAppListByFileName(nodeList, null);
+    }
+    
+    /**
+     * @param nodeList application node list to parse.
+     * @param fileName  application file name to search, 'null' otherwise. 
+     * @return true if the application with the file name is configured.
+     */
+    private static boolean isFoundAppConfigFromAppListByFileName(NodeList nodeList, String fileName) {
```

Comment:
line 329 has a todo for resolution of the variable in the 'location' element should be handled.  This one is important and I do not see an issue opened in the ecosystem github